ViAn/Video/frameprocessor.cpp

```diff
@@ -123,6 +125,21 @@ void FrameProcessor::process_frame() {
     m_overlay->draw_overlay(manipulated_frame, m_frame_index->load());
 
     // Scales the frame
+    cv::Rect prev_rect = m_zoomer.get_frame_rect();
+    if (!rotated && (prev_rect.width != m_width->load() || prev_rect.height != m_height->load())) {
+        // Source size has changed (image sequence)
+        m_zoomer.set_frame_size(cv::Size(m_width->load(), m_height->load()));
+        m_zoomer.fit_viewport();
+
+        // Store changes made
```

Comment:
Why is this done here? Isn't this done in the "update_zoomer_settings"?

Reply:
It is used to zoom out when images in a sequence are of varying sizes. So if the first image is very small and the second very large, this will make sure that the whole second image is displayed when "playing".